controllers/items.js

```diff
@@ -0,0 +1,151 @@
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items based on a particular item type
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnType = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnType(req.params.type);
+    return res.json({
+      message: "Items fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items based on a particular tag id
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnTagId = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnTagId(req.params.tagid);
+    return res.json({
+      message: "Items fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items with its tags and levels data based on a particular itemid
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnitemId = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnItemId(req.params.itemid);
+    return res.json({
+      message: "Item fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns items based on a particular tag type
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getItemsBasedOnTagType = async (req, res) => {
+  try {
+    const items = await ItemModel.getItemsBasedOnTagType(req.params.tagtype);
+    return res.json({
+      message: "Item fetched Successfully",
+      data: items,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new item: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+/**
+ * Collects and returns all items with some tag and level
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getAllItems = async (req, res) => {
+  try {
+    const allItems = await ItemModel.getAllItems();
+    return res.json({
+      allItems,
+    });
+  } catch (err) {
+    logger.error(`Error while creating new Level: ${err}`);
+    return res.boom.badImplementation("An internal server error occurred");
+  }
+};
+
+module.exports = {
```

Comment:
Return response formate is changing at all control functions. Make it consistent.